app/src/main/java/chat/rocket/android/api/rest/ServerPolicyApi.java

```diff
@@ -0,0 +1,19 @@
+package chat.rocket.android.api.rest;
+
+import android.support.annotation.NonNull;
+import org.json.JSONObject;
+
+public interface ServerPolicyApi {
+
+  void getApiInfoSecurely(@NonNull Callback callback);
+
+  void getApiInfoInsecurely(@NonNull Callback callback);
+
+  interface Callback {
```

Comment:
Also, I think `rx.Single` or  `bolts.Task` would make it easier to write the logic of sequential API calls or retrying, and actually prefer using them.
How do you think?